--- mate-notification-daemon-1.5.0/configure.ac.0001	2013-02-17 02:52:47.604932539 +0100
+++ mate-notification-daemon-1.5.0/configure.ac	2013-02-17 02:52:47.636932246 +0100
@@ -123,7 +123,7 @@ pkg_modules="
 	dbus-1 >= $REQ_DBUS_VERSION, \
 	dbus-glib-1 >= $REQ_DBUS_VERSION, \
         libcanberra-gtk >= $REQ_LIBCANBERRA_GTK_VERSION, \
-	libmatewnck \
+	libwnck-1.0 \
         x11 \
 "
 PKG_CHECK_MODULES(NOTIFICATION_DAEMON, $pkg_modules)
--- mate-notification-daemon-1.5.2/src/daemon/daemon.c~	2013-03-11 12:55:15.000000000 +0200
+++ mate-notification-daemon-1.5.2/src/daemon/daemon.c	2013-03-11 12:58:19.295122631 +0200
@@ -42,7 +42,8 @@
 #include <X11/Xatom.h>
 #include <gdk/gdkx.h>
 
-#include <libmatewnck/libmatewnck.h>
+#define WNCK_I_KNOW_THIS_IS_UNSTABLE
+#include <libwnck/libwnck.h>
 
 #include "daemon.h"
 #include "engines.h"
@@ -1148,36 +1148,36 @@
 
 static gboolean fullscreen_window_exists(GtkWidget* nw)
 {
-	MatewnckScreen* matewnck_screen;
-	MatewnckWorkspace* matewnck_workspace;
+	WnckScreen* matewnck_screen;
+	WnckWorkspace* matewnck_workspace;
 	GList* l;
 
 	#if (GDK_PIXBUF_MAJOR == 2) && (GDK_PIXBUF_MINOR < 22)
-		matewnck_screen = matewnck_screen_get(GDK_SCREEN_XNUMBER(gdk_window_get_screen(gtk_widget_get_window(nw))));
+		matewnck_screen = wnck_screen_get(GDK_SCREEN_XNUMBER(gdk_window_get_screen(gtk_widget_get_window(nw))));
 	#else
-		matewnck_screen = matewnck_screen_get(GDK_SCREEN_XNUMBER(gdk_drawable_get_screen(GDK_DRAWABLE(GTK_WIDGET(nw)->window))));
+		matewnck_screen = wnck_screen_get(GDK_SCREEN_XNUMBER(gdk_drawable_get_screen(GDK_DRAWABLE(GTK_WIDGET(nw)->window))));
 	#endif
 
-	matewnck_screen_force_update (matewnck_screen);
+	wnck_screen_force_update (matewnck_screen);
 
-	matewnck_workspace = matewnck_screen_get_active_workspace (matewnck_screen);
+	matewnck_workspace = wnck_screen_get_active_workspace (matewnck_screen);
 
-	for (l = matewnck_screen_get_windows_stacked (matewnck_screen); l != NULL; l = l->next)
+	for (l = wnck_screen_get_windows_stacked (matewnck_screen); l != NULL; l = l->next)
 	{
-		MatewnckWindow *matewnck_win = (MatewnckWindow *) l->data;
+		WnckWindow *matewnck_win = (WnckWindow *) l->data;
 
-		if (matewnck_window_is_on_workspace (matewnck_win, matewnck_workspace) && matewnck_window_is_fullscreen (matewnck_win) && matewnck_window_is_active (matewnck_win))
+		if (wnck_window_is_on_workspace (matewnck_win, matewnck_workspace) && wnck_window_is_fullscreen (matewnck_win) && wnck_window_is_active (matewnck_win))
 		{
 			/*
 			 * Sanity check if the window is _really_ fullscreen to
 			 * work around a bug in libmatewnck that doesn't get all
 			 * unfullscreen events.
 			 */
-			int sw = matewnck_screen_get_width (matewnck_screen);
-			int sh = matewnck_screen_get_height (matewnck_screen);
+			int sw = wnck_screen_get_width (matewnck_screen);
+			int sh = wnck_screen_get_height (matewnck_screen);
 			int x, y, w, h;
 
-			matewnck_window_get_geometry (matewnck_win, &x, &y, &w, &h);
+			wnck_window_get_geometry (matewnck_win, &x, &y, &w, &h);
 
 			if (sw == w && sh == h)
 			{
